### PR TITLE
Clarified local Docker image limitations

### PR DESCRIPTION
**Pull request by cwarnermm**
Local Docker image setup documentation updated based on feedback provided via docs.mattermost.com stating that docs aren't clear about local image limitations
- updated description to be more explicit and clear about the reasons why the local image shouldn't be used in prod
- emphasized the message that the local image shouldn't be used in prod
- added context around the fact that the local image has an internal DB/works out of the box/is self-contained
- updated LHS entry to clarify it's a local image

### CODE REVIEW

File: source/install/setting-up-local-machine-using-docker.rst [hunk 1]
```diff
@@ -15,10 +15,11 @@ Local Machine Setup using Docker
...
 
-.. important::
-  This configuration shouldn't be used in production, as it uses a known password string, contains other non-production configuration settings, keeps no persistent data (all data lives inside the container) and doesn't support upgrades. For a production installation with Docker, see the `Mattermost Docker Setup README <https://github.com/mattermost/docker#mattermost-docker-setup>`__.
+This local ``mattermost-preview`` image is a self-contained image (i.e., it has an internal database and works out of the box). Dropping a container using this image removes data and configuration as expected. It also uses a known password string, contains other non-production configuration settings, keeps no persistent data (all data lives inside the container) and doesn't support upgrades.
+
+**This image shouldn't be used in production environments.** For a production installation with Docker, see the `Mattermost Docker Setup README <https://github.com/mattermost/docker#mattermost-docker-setup>`__.
```
Comment (justinegeffen) [inline source/install/setting-up-local-machine-using-docker.rst:22]:
```suggestion
**This image shouldn't be used in production environments.** For a production installation with Docker, see the `Mattermost Docker Setup README <https://github.com/mattermost/docker#mattermost-docker-setup>`__.
```

**Review by justinegeffen — COMMENTED** [review]
A few suggestions.

File: source/guides/deployment.rst [hunk 2]
```diff
@@ -9,19 +9,19 @@ Get started
...
     Implement Mattermost </getting-started/implementation-plan>
     Enterprise roll out checklist </getting-started/enterprise-roll-out-checklist>
     Welcome email template </getting-started/welcome-email-to-end-users>
 
 These guides will get you up and running with Mattermost in minutes.
 
-* :doc:`Quick install guide </getting-started/light-install>` - Deploy in minutes via Mattermost Omnibus on Ubuntu.
-* :doc:`Run Mattermost via Docker </install/setting-up-local-machine-using-docker>` - Launch a Mattermost server instantly to test functionality and build integrations.
-* :doc:`Administrator tasks </getting-started/admin-onboarding-tasks>` - Learn about the standard configurations and settings you’ll encounter.
+* :doc:`Quick Install Guide </getting-started/light-install>` - Deploy in minutes via Mattermost Omnibus on Ubuntu.
```
Comment (justinegeffen) [inline source/guides/deployment.rst:22]: Thinking this should be "Quick install guide"?

File: source/guides/deployment.rst [hunk 3]
```diff
@@ -9,19 +9,19 @@ Get started
     :maxdepth: 1
     :hidden:
 
-    Quick install guide </getting-started/light-install>
-    Run Mattermost via Docker </install/setting-up-local-machine-using-docker>
-    Administrator tasks </getting-started/admin-onboarding-tasks>
+    Quick Install Guide </getting-started/light-install>
```
Comment (cwarnermm) [inline source/guides/deployment.rst:12]:
```suggestion
    Quick install guide </getting-started/light-install>
```

File: source/guides/deployment.rst [hunk 4]
```diff
@@ -9,19 +9,19 @@ Get started
...
+    Run Mattermost locally via Docker </install/setting-up-local-machine-using-docker>
+    Administrator Tasks </getting-started/admin-onboarding-tasks>
```
Comment (cwarnermm) [inline source/guides/deployment.rst:14]:
```suggestion
    Administrator tasks </getting-started/admin-onboarding-tasks>
```

File: source/guides/deployment.rst (hunk 2 above)
Comment (cwarnermm) [inline source/guides/deployment.rst:22]:
```suggestion
* :doc:`Quick install guide </getting-started/light-install>` - Deploy in minutes via Mattermost Omnibus on Ubuntu.
```

**Comment by azigler** [comment]
We're updating this page via #5517 to have both preview and production instructions. I connected with @cwarnermm about this! Once that is merged, we can close this PR. I kept the [Administrator tasks](https://docs.mattermost.com/getting-started/admin-onboarding-tasks.html) page for now, but I did improve the nav links so we're clearly directing users to the preview vs. production versions of Docker, where appropriate.

**Pull request closed without merging**